transformer_engine/pytorch/csrc/extensions.cu

```diff
@@ -1131,23 +1131,24 @@ at::Tensor fp8_transpose(at::Tensor input,
 }
 
 
-at::Tensor fp8_gelu(at::Tensor input,
-                    at::Tensor scale,
-                    at::Tensor amax,
-                    at::Tensor scale_inv,
-                    transformer_engine::DType otype
+at::Tensor gelu(at::Tensor input,
+                at::Tensor scale,
+                at::Tensor amax,
+                at::Tensor scale_inv,
+                transformer_engine::DType otype
 ) {
   using namespace transformer_engine;
 
-  size_t M = static_cast<size_t>(input.size(0));
-  size_t N = static_cast<size_t>(input.size(1));
+  size_t N = static_cast<size_t>(input.size(-1));
+  size_t M = numel(input) / N;
 
   auto output =
-            allocateTorchTensor(input.size(0),
-                                input.size(1),
-                                DType::kByte);
+            allocateTorchTensor(M,
+                                N,
+                                otype);
```

Comment:
I think it makes more sense if we return a Torch tensor with the same dims as the input. Internally we can treat it as 2D, or even 1D. We can either modify `allocateTorchTensor` to handle general tensor dims or do `return output.view(input.sizes())`.

For the GLU case, I think all dims but the last should be the same.

Reply:
Currently this is what is happening since our use case is purely 2D. If the suggestion is about a general change in the ability to handle other sized dims, then I think it falls outside the scope of this PR.